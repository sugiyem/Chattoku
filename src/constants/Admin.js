export const AdminPower = {
  DELETE_POSTS: "Delete Posts",
  BAN_USER: "Ban Users From Forum"
};
