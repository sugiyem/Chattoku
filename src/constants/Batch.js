export const MAXIMUM_BATCH_SIZE = 400;
