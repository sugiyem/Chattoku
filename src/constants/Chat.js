export const DEFAULT_AVATAR_URL =
  "https://firebasestorage.googleapis.com/v0/b/chattoku-e2672.appspot.com/" +
  "o/default-profile.png?alt=media&token=e05118be-75e9-42ed-8568-c42a3d34d6f1";

export const chatType = {
  PRIVATE_CHAT: 0,
  GROUP_CHAT: 1
};
