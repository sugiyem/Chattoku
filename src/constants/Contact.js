export const contactType = {
  USER: 0,
  GROUP: 1
};
