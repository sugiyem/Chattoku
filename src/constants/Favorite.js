export const favoriteType = {
  ANIME: 0,
  GENRE: 1
};
