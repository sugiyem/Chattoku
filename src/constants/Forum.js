export const renderType = {
  CREATE: 0,
  EDIT: 1
};

export const manageType = {
  BAN: 0,
  ADMIN: 1
};
