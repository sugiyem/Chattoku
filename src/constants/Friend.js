export const friendshipType = {
  FRIEND: 0,
  WAITING_RESPONSE: 1,
  RECEIVING_REQUEST: 2,
  NON_FRIEND: 3,
  BLOCKED: 4
};
