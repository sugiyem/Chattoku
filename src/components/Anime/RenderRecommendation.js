import React, { useEffect, useState } from "react";
import { Alert, StyleSheet, View } from "react-native";
import {
  Button,
  ButtonText,
  CenteredBoldText,
  ScrollContainer,
  SeparatedButton
} from "../../styles/GeneralStyles";
import AnimeCard from "./AnimeCard";
import FetchRecommendations from "../../services/Anime/FetchRecommendations";
import FetchFavoriteAnime from "../../services/Anime/FetchFavoriteAnime";
import Loading from "../Miscellaneous/Loading";
import { fetchType } from "../../constants/MyAnimeList";
import { favoriteType } from "../../constants/Favorite";
import {
  favoriteGenreListToGenreString,
  animeDetailsToGenreString
} from "../../services/Anime/AnimeAndGenreConverter";

const RenderRecommendation = ({
  type,
  genreList,
  sourceAnime = null,
  navigation
}) => {
  const [recommendations, setRecommendations] = useState([]);
  const [recommendationCount, setRecommendationCount] = useState(undefined);
  const [animeData, setAnimeData] = useState(null);
  const [favoriteAnimeID, setFavoriteAnimeID] = useState([]);
  const [isRecommendationFound, setIsRecommendationFound] = useState(false);
  const [isAnimeDataLoaded, setIsAnimeDataLoaded] = useState(true);

  const isCalledFromFavoriteAnime = type === favoriteType.ANIME;

  const increaseCount = () => {
    setRecommendationCount(recommendationCount + 1);
  };

  const decreaseCount = () => {
    setRecommendationCount(recommendationCount - 1);
  };

  useEffect(() => {
    return FetchFavoriteAnime({
      onSuccesfulFetch: (data) => {
        setFavoriteAnimeID(data.map((item) => item.id));
      },
      onFailure: (error) => {
        if (error.message === "canceled") {
          return;
        }
        Alert.alert("Error", error.message);
      }
    });
  }, []);

  useEffect(() => {
    const abortController = new AbortController();
    setIsRecommendationFound(false);

    FetchRecommendations({
      genresString: isCalledFromFavoriteAnime
        ? animeDetailsToGenreString(sourceAnime)
        : favoriteGenreListToGenreString(genreList),
      onSuccess: (data) => {
        if (isCalledFromFavoriteAnime) {
          data = data.filter((item) => item != sourceAnime.id.toString());
        }
        setRecommendations(data);
        setIsRecommendationFound(true);
      },
      abortController: abortController
    });

    return () => abortController.abort();
  }, []);

  useEffect(() => {
    if (recommendations.length > 0) {
      setRecommendationCount(0);
    }
  }, [recommendations]);

  useEffect(() => {
    if (recommendationCount === undefined) {
      setAnimeData(null);
      return;
    }

    const abortController = new AbortController();
    setIsAnimeDataLoaded(false);

    AnimeFetch({
      type: fetchType.BY_ID,
      malID: recommendations[recommendationCount],
      onSuccesfulFetch: (data) => {
        setAnimeData(data);
        setIsAnimeDataLoaded(true);
      },
      abortController: abortController
    });

    return () => abortController.abort();
  }, [recommendations, recommendationCount]);

  const isPageLoading = !(isRecommendationFound && isAnimeDataLoaded);

  const RenderPage = () => (
    <ScrollContainer>
      <CenteredBoldText size="20px">
        {isCalledFromFavoriteAnime
          ? "Anime similar to " + sourceAnime.title
          : "Anime which has similar theme to your favorite genres"}
      </CenteredBoldText>

      <Button onPress={() => navigation.goBack()} testID="backButton">
        <ButtonText>Go Back</ButtonText>
      </Button>

      <View style={styles.buttonGroup}>
        {recommendationCount > 0 && (
          <SeparatedButton onPress={decreaseCount} testID="decreaseButton">
            <ButtonText size="12px">Previous recommendation</ButtonText>
          </SeparatedButton>
        )}
        {recommendationCount < recommendations.length - 1 && (
          <SeparatedButton onPress={increaseCount} testID="increaseButton">
            <ButtonText size="12px">Next recommendation</ButtonText>
          </SeparatedButton>
        )}
      </View>

      {animeData && (
        <AnimeCard
          item={animeData}
          isFavorite={favoriteAnimeID.includes(animeData.mal_id)}
        />
      )}
    </ScrollContainer>
  );

  return (
    <Loading isLoading={isPageLoading}>
      <RenderPage />
    </Loading>
  );
};

export default RenderRecommendation;

const styles = StyleSheet.create({
  buttonGroup: {
    paddingBottom: 10,
    borderBottomWidth: 1,
    flexDirection: "row"
  }
});
