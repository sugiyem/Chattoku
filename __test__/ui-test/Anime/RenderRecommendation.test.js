import React from "react";
import renderer from "react-test-renderer";
import { fireEvent, render } from "@testing-library/react-native";
import RenderRecommendation from "../../../src/components/Anime/RenderRecommendation";
import { favoriteType } from "../../../src/constants/Favorite";

const mockSetState = jest.fn();
const mockBackNavigate = jest.fn();
const mockNavigation = {
  goBack: mockBackNavigate
};
const mockSource = {
  demographics: [],
  genres: ["Action", "Adventure", "Fantasy"],
  id: 23321,
  image: "https://cdn.myanimelist.net/images/anime/5/68097.jpg",
  themes: ["Adult Cast", "Video Game"],
  title: "Log Horizon 2nd Season",
  url: "https://myanimelist.net/anime/23321/Log_Horizon_2nd_Season"
};
const mockData = {
  aired: {
    from: "2013-10-05T00:00:00+00:00",
    prop: {
      from: {
        day: 5,
        month: 10,
        year: 2013
      },
      to: {
        day: 22,
        month: 3,
        year: 2014
      }
    },
    string: "Oct 5, 2013 to Mar 22, 2014",
    to: "2014-03-22T00:00:00+00:00"
  },
  airing: false,
  background:
    "Log Horizon adapts the first five volumes of Mamare Touno's novel series of the same title.",
  broadcast: {
    day: "Saturdays",
    string: "Saturdays at 17:30 (JST)",
    time: "17:30",
    timezone: "Asia/Tokyo"
  },
  demographics: [],
  duration: "25 min per ep",
  episodes: 25,
  explicit_genres: [],
  favorites: 9950,
  genres: [
    {
      mal_id: 1,
      name: "Action",
      type: "anime",
      url: "https://myanimelist.net/anime/genre/1/Action"
    },
    {
      mal_id: 2,
      name: "Adventure",
      type: "anime",
      url: "https://myanimelist.net/anime/genre/2/Adventure"
    },
    {
      mal_id: 10,
      name: "Fantasy",
      type: "anime",
      url: "https://myanimelist.net/anime/genre/10/Fantasy"
    }
  ],
  images: {
    jpg: {
      image_url: "https://cdn.myanimelist.net/images/anime/5/84004.jpg",
      large_image_url: "https://cdn.myanimelist.net/images/anime/5/84004l.jpg",
      small_image_url: "https://cdn.myanimelist.net/images/anime/5/84004t.jpg"
    },
    webp: {
      image_url: "https://cdn.myanimelist.net/images/anime/5/84004.webp",
      large_image_url: "https://cdn.myanimelist.net/images/anime/5/84004l.webp",
      small_image_url: "https://cdn.myanimelist.net/images/anime/5/84004t.webp"
    }
  },
  licensors: [
    {
      mal_id: 102,
      name: "Funimation",
      type: "anime",
      url: "https://myanimelist.net/anime/producer/102/Funimation"
    },
    {
      mal_id: 376,
      name: "Sentai Filmworks",
      type: "anime",
      url: "https://myanimelist.net/anime/producer/376/Sentai_Filmworks"
    }
  ],
  mal_id: 17265,
  members: 1027837,
  popularity: 117,
  producers: [
    {
      mal_id: 111,
      name: "NHK",
      type: "anime",
      url: "https://myanimelist.net/anime/producer/111/NHK"
    }
  ],
  rank: 662,
  rating: "PG-13 - Teens 13 or older",
  score: 7.94,
  scored_by: 539375,
  season: "fall",
  source: "Light novel",
  status: "Finished Airing",
  studios: [
    {
      mal_id: 41,
      name: "Satelight",
      type: "anime",
      url: "https://myanimelist.net/anime/producer/41/Satelight"
    }
  ],
  synopsis:
    'In the blink of an eye, thirty thousand bewildered Japanese gamers are whisked from their everyday lives into the world of the popular MMORPG, Elder Tale, after the game\'s latest update—unable to log out. Among them is the socially awkward college student Shiroe, whose confusion and shock lasts only a moment as, a veteran of the game, he immediately sets out to explore the limits of his new reality. Shiroe must learn to live in this new world, leading others and negotiating with the NPC "natives" in order to bring stability to the virtual city of Akihabara. He is joined by his unfortunate friend Naotsugu, having logged in for the first time in years only to find himself trapped, and Akatsuki, a petite but fierce assassin who labels Shiroe as her master. A tale of fantasy, adventure, and politics, Log Horizon explores the elements of gaming through the eyes of a master strategist who attempts to make the best of a puzzling situation. [Written by MAL Rewrite]',
  themes: [
    {
      mal_id: 50,
      name: "Adult Cast",
      type: "anime",
      url: "https://myanimelist.net/anime/genre/50/Adult_Cast"
    },
    {
      mal_id: 79,
      name: "Video Game",
      type: "anime",
      url: "https://myanimelist.net/anime/genre/79/Video_Game"
    }
  ],
  title: "Log Horizon",
  title_english: "Log Horizon",
  title_japanese: "ログ・ホライズン",
  title_synonyms: [],
  trailer: {
    embed_url:
      "https://www.youtube.com/embed/IG1VhJ75r8k?enablejsapi=1&wmode=opaque&autoplay=1",
    images: {
      image_url: "https://img.youtube.com/vi/IG1VhJ75r8k/default.jpg",
      large_image_url: "https://img.youtube.com/vi/IG1VhJ75r8k/hqdefault.jpg",
      maximum_image_url:
        "https://img.youtube.com/vi/IG1VhJ75r8k/maxresdefault.jpg",
      medium_image_url: "https://img.youtube.com/vi/IG1VhJ75r8k/mqdefault.jpg",
      small_image_url: "https://img.youtube.com/vi/IG1VhJ75r8k/sddefault.jpg"
    },
    url: "https://www.youtube.com/watch?v=IG1VhJ75r8k",
    youtube_id: "IG1VhJ75r8k"
  },
  type: "TV",
  url: "https://myanimelist.net/anime/17265/Log_Horizon",
  year: 2013
};
jest.mock("@react-navigation/core", () => ({
  useNavigation: () => mockNavigation
}));

jest.mock("react-native/Libraries/Animated/NativeAnimatedHelper");
jest.mock("react", () => ({
  ...jest.requireActual("react"),
  useState: (initial) => [
    initial === undefined
      ? 1
      : initial === null
      ? mockData
      : typeof initial === "boolean"
      ? true
      : [1, 2, 3],
    mockSetState
  ],
  useEffect: () => {}
}));

describe("RenderRecommendation UI test", () => {
  it("renders correctly", () => {
    const tree = renderer
      .create(
        <RenderRecommendation
          type={favoriteType.ANIME}
          genreList={[]}
          sourceAnime={mockSource}
          navigation={mockNavigation}
        />
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it("Navigation Button Works", () => {
    const { getByTestId } = render(
      <RenderRecommendation
        type={favoriteType.ANIME}
        genreList={[]}
        sourceAnime={mockSource}
        navigation={mockNavigation}
      />
    );

    fireEvent.press(getByTestId("backButton"));
    expect(mockBackNavigate).toHaveBeenCalled();
  });

  it("Pagination Button Works", () => {
    const { getByTestId } = render(
      <RenderRecommendation
        type={favoriteType.ANIME}
        genreList={[]}
        sourceAnime={mockSource}
        navigation={mockNavigation}
      />
    );

    fireEvent.press(getByTestId("decreaseButton"));
    expect(mockSetState).toHaveBeenCalledWith(0);

    fireEvent.press(getByTestId("increaseButton"));
    expect(mockSetState).toHaveBeenCalledWith(2);
  });
});
