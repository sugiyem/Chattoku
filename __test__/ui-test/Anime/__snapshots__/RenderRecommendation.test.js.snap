// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RenderRecommendation UI test renders correctly 1`] = `
<ViewManagerAdapter_ExpoLinearGradient
  proxiedProperties={
    Object {
      "colors": Array [
        4287090411,
        4283256319,
      ],
      "endPoint": Array [
        1,
        1,
      ],
      "locations": undefined,
      "startPoint": Array [
        0,
        0,
      ],
    }
  }
  style={
    Object {
      "flex": 1,
      "padding": 5,
    }
  }
>
  <RCTScrollView
    style={
      Object {
        "flex": 1,
        "padding": 5,
      }
    }
  >
    <View>
      <Text
        size="20px"
        style={
          Array [
            Object {
              "fontFamily": "Gill Sans",
              "fontSize": 20,
              "fontWeight": "bold",
              "textAlign": "center",
              "textDecorationLine": "none",
            },
          ]
        }
      >
        Anime similar to Log Horizon 2nd Season
      </Text>
      <View
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignSelf": "stretch",
            "backgroundColor": "#7FFFD4",
            "borderRadius": 10,
            "borderWidth": 1,
            "marginBottom": 5,
            "marginLeft": 5,
            "marginRight": 5,
            "marginTop": 5,
            "opacity": 1,
            "paddingBottom": 5,
            "paddingLeft": 5,
            "paddingRight": 5,
            "paddingTop": 5,
          }
        }
        testID="backButton"
      >
        <Text
          style={
            Array [
              Object {
                "color": "#483d8b",
                "fontSize": 14,
                "textAlign": "center",
              },
            ]
          }
        >
          Go Back
        </Text>
      </View>
      <View
        style={
          Object {
            "borderBottomWidth": 1,
            "flexDirection": "row",
            "paddingBottom": 10,
          }
        }
      >
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignSelf": "stretch",
              "backgroundColor": "#00ffff",
              "borderColor": "#000000",
              "borderRadius": 10,
              "borderWidth": 1,
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "marginBottom": 5,
              "marginHorizontal": 10,
              "marginLeft": 5,
              "marginRight": 5,
              "marginTop": 5,
              "opacity": 1,
              "paddingBottom": 5,
              "paddingLeft": 5,
              "paddingRight": 5,
              "paddingTop": 5,
            }
          }
          testID="decreaseButton"
        >
          <Text
            size="12px"
            style={
              Array [
                Object {
                  "color": "#483d8b",
                  "fontSize": 12,
                  "textAlign": "center",
                },
              ]
            }
          >
            Previous recommendation
          </Text>
        </View>
        <View
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignSelf": "stretch",
              "backgroundColor": "#00ffff",
              "borderColor": "#000000",
              "borderRadius": 10,
              "borderWidth": 1,
              "flexBasis": 0,
              "flexGrow": 1,
              "flexShrink": 1,
              "marginBottom": 5,
              "marginHorizontal": 10,
              "marginLeft": 5,
              "marginRight": 5,
              "marginTop": 5,
              "opacity": 1,
              "paddingBottom": 5,
              "paddingLeft": 5,
              "paddingRight": 5,
              "paddingTop": 5,
            }
          }
          testID="increaseButton"
        >
          <Text
            size="12px"
            style={
              Array [
                Object {
                  "color": "#483d8b",
                  "fontSize": 12,
                  "textAlign": "center",
                },
              ]
            }
          >
            Next recommendation
          </Text>
        </View>
      </View>
      <View
        style={
          Array [
            Object {
              "backgroundColor": "white",
              "borderRadius": 10,
              "marginBottom": 20,
              "marginLeft": 5,
              "marginRight": 5,
              "marginTop": 20,
              "paddingBottom": 15,
              "paddingLeft": 15,
              "paddingRight": 15,
              "paddingTop": 15,
            },
          ]
        }
      >
        <View
          style={
            Array [
              Object {
                "alignItems": "center",
                "flexDirection": "row",
                "marginBottom": 20,
                "width": "100%",
              },
            ]
          }
        >
          <Image
            source={
              Object {
                "uri": "https://cdn.myanimelist.net/images/anime/5/84004.jpg",
              }
            }
            style={
              Array [
                Object {
                  "height": 300,
                  "width": 225,
                },
              ]
            }
          />
          <View
            style={
              Array [
                Object {
                  "flexBasis": 0,
                  "flexGrow": 1,
                  "flexShrink": 1,
                  "height": "100%",
                  "marginLeft": 20,
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "fontSize": 18,
                    "fontWeight": "500",
                    "textAlign": "center",
                    "width": "100%",
                  },
                ]
              }
              testID="title"
            >
              Log Horizon
            </Text>
            <View
              style={
                Array [
                  Object {
                    "backgroundColor": "black",
                    "height": 1,
                    "marginBottom": 5,
                    "marginLeft": 0,
                    "marginRight": 0,
                    "marginTop": 5,
                  },
                ]
              }
            />
            <Text
              style={
                Array [
                  Object {
                    "fontSize": 17,
                    "fontWeight": "500",
                    "marginBottom": 5,
                    "marginLeft": 0,
                    "marginRight": 0,
                    "marginTop": 5,
                  },
                ]
              }
            >
              Genres: 
            </Text>
            <Text
              testID="genres"
            >
              Action, Adventure, Fantasy, Adult Cast, Video Game
            </Text>
          </View>
        </View>
        <View>
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "navy",
                "borderColor": "lightblue",
                "borderRadius": 15,
                "borderWidth": 2,
                "marginBottom": 10,
                "marginLeft": 0,
                "marginRight": 0,
                "marginTop": 10,
                "opacity": 1,
                "paddingBottom": 5,
                "paddingLeft": 5,
                "paddingRight": 5,
                "paddingTop": 5,
              }
            }
            testID="editFavorite"
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "lightblue",
                    "fontSize": 16,
                    "textAlign": "center",
                  },
                ]
              }
            >
              Add this anime to favorite
            </Text>
          </View>
          <Text
            size="20px"
            style={
              Array [
                Object {
                  "fontSize": 17,
                  "fontWeight": "500",
                  "marginBottom": 5,
                  "marginLeft": 0,
                  "marginRight": 0,
                  "marginTop": 5,
                },
              ]
            }
          >
            Synopsis:
          </Text>
          <Text
            style={
              Array [
                Object {
                  "fontSize": 14,
                },
              ]
            }
            testID="synopsis"
          >
            In the blink of an eye, thirty thousand bewildered Japanese gamers are whisked from their everyday lives into the world of the popular MMORPG, Elder Tale, after the game's latest update—unable to log out. Among them is the socially awkward college student Shiroe, whose confusion and shock lasts only a moment as, a veteran of the game, he immediately sets out to explore the limits of his new reality. Shiroe must learn to live in this new world, leading others and negotiating with the NPC "natives" in order to bring stability to the virtual city of Akihabara. He is joined by his unfortunate friend Naotsugu, having logged in for the first time in years only to find himself trapped, and Akatsuki, a petite but fierce assassin who labels Shiroe as her master. A tale of fantasy, adventure, and politics, Log Horizon explores the elements of gaming through the eyes of a master strategist who attempts to make the best of a puzzling situation. [Written by MAL Rewrite]
          </Text>
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "navy",
                "borderColor": "lightblue",
                "borderRadius": 15,
                "borderWidth": 2,
                "marginBottom": 10,
                "marginLeft": 0,
                "marginRight": 0,
                "marginTop": 10,
                "opacity": 1,
                "paddingBottom": 5,
                "paddingLeft": 5,
                "paddingRight": 5,
                "paddingTop": 5,
              }
            }
            testID="openURL"
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "lightblue",
                    "fontSize": 16,
                    "textAlign": "center",
                  },
                ]
              }
            >
              Click here for more information!
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</ViewManagerAdapter_ExpoLinearGradient>
`;
